src/thinblock.cpp

```diff
@@ -119,17 +119,24 @@ bool CThinBlock::process(CNode *pfrom, int nSizeThinBlock, string strCommand)
             }
             if (tx.IsNull())
                 missingCount++;
-            // This will push an empty/invalid transaction if we don't have it yet
-            pfrom->thinBlock.vtx.push_back(tx);
 
             // In order to prevent a memory exhaustion attack we track transaction bytes used to create Block
             // to see if we've exceeded any limits and if so clear out data and return.
             uint64_t nTxSize = RecursiveDynamicUsage(tx);
-            if (thindata.AddThinBlockBytes(nTxSize, pfrom) > maxAllowedSize)
+            if (thindata.AddThinBlockBytes(nTxSize, pfrom) > maxAllowedSize-nTxSize)
```

Comment:
this will probably hiccup on travis ... run clang-fomat on this or put spaces around the "-"